s1denoise/sentinel1image.py

```diff
@@ -434,6 +434,35 @@ def get_raw_sigma0_vectors_from_full_size(self, polarization, line, pixel, sigma
             raw_sigma0[i] = np.nanmean(sigma0_fs[int(y0):int(y1)], axis=0)[pixel[i]]
         return raw_sigma0
 
+    def get_lons_vectors_from_full_size(self, line, pixel, lons_fs):
+        """ Read lons from input GeoTIff for given lines and for given pixels
+            from full size longitude matrix
+
+        """
+        lons_res = [np.zeros(p.size)+np.nan for p in pixel]
+        for i in range(line.shape[0]):
+            lons_res[i] = lons_fs[i][pixel[i]]
```

Comment:
It seems that you will read not from the input `lines` but from the first N-lines (N=`line.shape[0]`). 

I think it should be:
```suggestion
            lons_res[i] = lons_fs[line[i]][pixel[i]]
```

Also below.

Reply:
I fully agree with your suggestions.